apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: 'pgvector: Fewer dimensions are better'
+description: Increase performance in pgvector by using embedding vectors with fewer dimensions
+author: gregnr,oli_rice,egor_romanov
+tags:
+  - AI
+  - performance
+  - postgres
+  - planetpg
+date: '2023-08-02'
+toc_depth: 3
+image: 2023-08-03-fewer-dimensions-are-better-pgvector/fewer-dimensions-og.png
+thumb: 2023-08-03-fewer-dimensions-are-better-pgvector/fewer-dimensions-thumb.png
+---
+
+Embeddings have become a core tool in AI applications to determine semantic similarity between pieces of content. Developers are using embeddings for various tasks, such as search, retrieval augmented generation, and clustering.
+
+At Supabase, we support storing embeddings in Postgres using the [pgvector](https://github.com/pgvector/pgvector) extension. pgvector adds a new datatype `vector`, which developers use to store embeddings in regular columns within their database tables. Similarity can be calculated between these vectors using one of the three supported distance measures:
+
+- Inner product
+- Cosine distance
+- Euclidean distance
+
+## Challenges with pgvector
+
+Without indexes, pgvector performs a full table scan when you run a similarity query. This means distance has to be computed against every row in your table. This is manageable at a small scale, but becomes problematic as your table grows.
+
+To solve this, pgvector offers indexes. Indexes reorganize the data into data structures that exploit internal structure and enable approximate similarity search without referring to every record. Currently, pgvector supports an IVF index, with HNSW expected in the next release.
+
+IVF indexes work by clustering vectors into `lists`, and then querying only vectors within the same list (or multiple nearby lists, depending on the value of `probes`). To learn more about pgvector's IVF indexes and its different configurations, read the docs on [Managing indexes](https://supabase.com/docs/guides/ai/managing-indexes).
```

Comment:
> To learn more about pgvector's IVF indexes and its different configurations, read the docs on Managing indexes

I'd probably remove this (it breaks the flow). If you want to link to the page make it more "inline" - eg:

```
IVF [indexes](https://supabase.com/docs/guides/ai/managing-indexes) work by clustering vectors into `lists`, and then querying only vectors within the same list (or multiple nearby lists, depending on the value of `probes`).
```